pkg/controllers/user/alert/manager/prometheus_operator.go

```diff
@@ -85,13 +86,34 @@ func (c *PromOperatorCRDManager) SyncPrometheusRule(promRule *monitoringv1.Prome
 		return fmt.Errorf("get namespace %s for prometheus rule %s:%s failed, %v", ns.Name, promRule.Namespace, promRule.Name, err)
 	}
 
+	//sort before compare
+	var sortedGroups []monitoringv1.RuleGroup
+	for _, v := range promRule.Spec.Groups {
+		sortedRules := v.Rules
+		sort.Slice(sortedRules, func(i, j int) bool {
+			return sortedRules[i].Labels["rule_id"] < sortedRules[j].Labels["rule_id"]
+		})
+		group := v.DeepCopy()
+		group.Rules = sortedRules
+		sortedGroups = append(sortedGroups, *group)
+	}
+
+	sort.Slice(sortedGroups, func(i, j int) bool {
+		return sortedGroups[i].Name < sortedGroups[j].Name
+	})
+	promRule.Spec.Groups = sortedGroups
```

Comment:
`sort` function won't change the point of the slice. I think you can sort the group first. And sort the rules in each group.